src/interface/report/PhaseParser.ts

```diff
@@ -0,0 +1,91 @@
+import React, { ReactNode } from 'react';
+
+import { EventType, PhaseEvent } from 'parser/core/Events';
+import { findByBossId, Phase } from 'raids';
+
+export const SELECTION_ALL_PHASES = "ALL";
+export const SELECTION_CUSTOM_PHASE = "CUSTOM";
+
+interface Props {
+  fight: {
+    // eslint-disable-next-line camelcase
+    start_time: number,
+    // eslint-disable-next-line camelcase
+    end_time: number,
+    boss: number,
+  },
+  bossPhaseEvents: PhaseEvent[],
+  children: (isLoading: boolean, phases?: {[key:string]: Phase}) => ReactNode,
+}
+
+interface State {
+  isLoading: boolean,
+  phases?: {[key:string]: Phase},
```

Comment:
```suggestion
  phases: { [key:string]: Phase } | undefined,
```

I believe propName**?** is intended for optional props, while in this case the prop will always exist, it may just have an undefined value. Not sure though, but it may still be worthwhile to make the distinction.